定义 User 和 Record 接口

tortoise014 opened the pull request.

tortoise014 committed:
登陆,注册,加上异常集中处理,mybatis和mybatisplus不兼容好像是

suyiiyii committed:
更新hutool依赖并移除mybatis-starter

此更改升级了hutool版本从5.8.11到5.8.26，并且移除了mybatis-spring-boot-starter依赖。这对于优化项目依赖和保持项目轻量级有好处。

tortoise014 committed:
数据库表名有点小问题

tortoise014 committed:
数据库表名有点小问题

suyiiyii committed:
解决用户表映射问题以匹配实际数据库结构

之前，用户表列名在MyBatis映射中被错误地标注为`userId`和`group`，而数据库中的实际列名为`user_id`和`group`（无反引号）。此次更改通过在MyBatis注释中使用正确的列名来解决了该问题，确保了数据操作的正确执行。此外，还修正了选择语句中的列名，以确保查询能够正确执行并返回期望的结果。

suyiiyii commented:
仔细看一下[1a602df](https://github.com/suyiiyii/SIMS/pull/9/commits/1a602dfd1d8fc07faa03814c5cec922ae58a5a3d)这个提交，注意一下sql的语法

tortoise014 committed:
不会测那个apifox有token怎么加进去啊

tortoise014 committed:
拦截器ok

suyiiyii commented on src/main/java/top/suyiiyii/sims/common/JwtInterceptor.java:
```diff
@@ -0,0 +1,76 @@
+        if (user == null) {
+            throw new ServiceException("401", "请登录");
+        }
+        JWTVerifier jwtVerifier = JWT.require(Algorithm.HMAC256(user.getPassword())).build();//加密,认证
```
jwt的验证部分抽出来，做一个jwt utils

suyiiyii commented on src/main/java/top/suyiiyii/sims/utils/TokenUtils.java:
```diff
@@ -0,0 +1,65 @@
+     * @return: java.lang.String
+     */
+    public static String createToken(String userId, String sign) {
+        return JWT.create().withAudience(userId)
```
这里也是，生成jwt的逻辑抽出来，加到jwt utils里面

suyiiyii committed:
允许接口文档目录的未登录访问
启用自动建表

suyiiyii commented on src/main/java/top/suyiiyii/sims/controller/UserController.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    @PostMapping("/login")
+    public Result login(@RequestBody User user){
```
一个登陆接口，不需要怎么多信息，应该再建一个DTO，专门用来承载登陆请求。其他接口同理

tortoise014 committed:
jwtUtils

tortoise014 committed:
权限校验1

tortoise014 committed:
Merge remote-tracking branch 'origin/wr' into wr

suyiiyii commented on src/main/java/top/suyiiyii/sims/controller/UserController.java:
```diff
@@ -0,0 +1,135 @@
+        }
+        userService.updatePassword(user);
+        return Result.success("更新成功");
+    }
```
不是，哥们
这里就传过来一个密码，怎么判断原密码和更新新密码的

tortoise014 replied:
okok
我看看

tortoise014 committed:
管理员查看用户信息

suyiiyii commented on src/main/java/top/suyiiyii/sims/VO/UserVO.java:
```diff
@@ -0,0 +1,24 @@
+package top.suyiiyii.sims.VO;
```
不用搞VO吧，我们小项目，整个entity和DTO就差不多了

tortoise014 replied:
我可能没有太明白哈哈哈我去学学

suyiiyii commented on src/main/java/top/suyiiyii/sims/entity/User.java:
```diff
@@ -16,14 +20,20 @@
+    @TableField(exist = false)
+    private String token;
+    @TableField(exist = false)
+    private Set<Permissions> permissions;
```
这里之前加上的，不用了记得删掉

tortoise014 replied:
这个是用了DTO之后就可以不用这个咯

suyiiyii commented on src/main/java/top/suyiiyii/sims/service/UserService.java:
```diff
@@ -0,0 +1,145 @@
+    }
+    public List<UserVO> findAllUsers(){
+        List<User> users = userMapper.selectAll();
+        List<UserVO> userVOS = new ArrayList<>();
```
这里的VO都换成DTO

tortoise014 replied:
okok晚上搞

tortoise014 committed:
816

suyiiyii committed:
refactor(user): 简化控制器和实体类并调整注释格式

- 合并登录和注册请求模型，统一参数处理- 删除未使用的字段和导入语句，优化代码
-调整方法注释格式，提升可读性和一致性
- 实体类 `User` 移除 `name` 字段，避免冗余

BREAKING CHANGE: `User` 实体类移除 `name` 字段，可能影响依赖此字段的数据库查询。请确保更新相关代码以适应这一变更。

suyiiyii committed:
feat(user): 完成User的接口

suyiiyii committed:
feat(user): 完成Record的接口
